File: app/src/main/java/at/shockbytes/corey/ui/activity/SignUpActivity.java
package at.shockbytes.corey.ui.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import at.shockbytes.corey.R;
import at.shockbytes.corey.ui.fragment.SignUpFragment;

public class SignUpActivity extends AppCompatActivity {

    public static Intent newIntent(Context context) {
        return new Intent(context, SignUpActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        getSupportFragmentManager()
                .beginTransaction()
                .replace(android.R.id.content, SignUpFragment.Companion.newInstance())
                .commit();
    }
}

File: app/src/main/java/at/shockbytes/corey/ui/activity/SignUpActivity.kt
package at.shockbytes.corey.ui.activity

import android.os.Bundle
import androidx.fragment.app.Fragment
import at.shockbytes.core.ui.activity.base.ContainerBackNavigableActivity
import at.shockbytes.corey.R
import at.shockbytes.corey.dagger.AppComponent
import at.shockbytes.corey.ui.fragment.SignUpFragment

class SignUpActivity : ContainerBackNavigableActivity<AppComponent>() {

    override fun bindViewModel() = Unit
    override fun unbindViewModel() = Unit

    override val displayFragment: Fragment
        get() = SignUpFragment.newInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.AppTheme_Signup)
        super.onCreate(savedInstanceState)
    }
}
